exporter/exporterhelper/queued_retry_inmemory.go

```diff
@@ -71,49 +71,140 @@ func (qCfg *QueueSettings) Validate() error {
 	return nil
 }
 
+var (
+	errNoStorageClient        = errors.New("no storage client extension found")
+	errMultipleStorageClients = errors.New("multiple storage extensions found")
+)
+
 type queuedRetrySender struct {
-	fullName        string
-	cfg             QueueSettings
-	consumerSender  requestSender
-	queue           internal.ProducerConsumerQueue
-	retryStopCh     chan struct{}
-	traceAttributes []attribute.KeyValue
-	logger          *zap.Logger
+	fullName           string
+	id                 config.ComponentID
+	signal             config.DataType
+	cfg                QueueSettings
+	consumerSender     requestSender
+	queue              internal.ProducerConsumerQueue
+	retryStopCh        chan struct{}
+	traceAttributes    []attribute.KeyValue
+	logger             *zap.Logger
+	requeuingEnabled   bool
+	requestUnmarshaler internal.RequestUnmarshaler
 }
 
-func newQueuedRetrySender(id config.ComponentID, _ config.DataType, qCfg QueueSettings, rCfg RetrySettings, _ internal.RequestUnmarshaler, nextSender requestSender, logger *zap.Logger) *queuedRetrySender {
+func buildFullName(persistentQueueEnabled bool, name string, signal config.DataType) string {
+	if !persistentQueueEnabled || signal == "" {
+		return name
+	}
+	// For persistent queue, we must use signal type, since the full name is used to identify storage
+	// (which must be separate for each signal type)
+	return fmt.Sprintf("%s-%s", name, signal)
+}
```

Comment:
I don't believe this is the right thing to do... I think we should have 2 different labels one for signal one for name instead of concatenating. As I mentioned, we should have this in a separate PR and debate there.

Reply:
Yeah, I see your point. The current design is unexpectedly changing the label used for the metric. I will fix this